### ISSUE TITLE
Bucket types no longer work using the 2.5.2 client against riak TS 1.3 [JIRA: CLIENTS-846]

### ISSUE THREAD

**Issue by point9repeating** [basho/riak-python-client#471]
I've built a dev cluster using the TS 1.3.0.1 image for ubuntu 14.04 found here:
http://s3.amazonaws.com/downloads.basho.com/riak_ts/1.3/1.3.0/ubuntu/trusty/riak-ts_1.3.0-1_amd64.deb

The following raises "NotImplementedError: Server does not support bucket-types":

```python
import riak
nodes = [{"host": "127.0.0.1", "http_port": 47001, "pb_port": 47101}]
client = riak.RiakClient(nodes=nodes)
client.bucket_type("grid-point").bucket("foo").get("anything")
```

**Comment by point9repeating** [comment]
FYI, when we dug into the [feature detection](https://github.com/basho/riak-python-client/blob/0d2855cea1c077bad8be8c45ea742388f491d72a/riak/transports/feature_detect.py),  it looks like the client expects to see a version > 2.0. 

But, my client is seeing 0.8.0 as the server version:
```python
In [5]: with client._transport() as transport:
   ...:     print transport.get_server_info()
   ...:     
{'node': 'riak@192.168.0.6', 'server_version': '0.8.0'}
```

**Comment by lukebakken** [comment]
This is an issue with Riak TS. I'll put a workaround in the Python client. As an initial workaround, you can hard-code the server version returned here to be `2.1.4`:

https://github.com/basho/riak-python-client/blob/master/riak/transports/tcp/transport.py#L78-L81

**Comment by bobhall** [comment]
Interestingly, I can't reproduce this same issue but I installed Riak 1.3 using riak-ts-ee_1.3.0-1_amd64.deb.

**Comment by lukebakken** [comment]
@bobhall - if you have a sec, could you run the code that @point9repeating provided above and post the output of `get_server_info()` here? Thanks! :smile: :fish_cake:

**Comment by bobhall** [comment]
`(Pdb) with client._transport() as transport: print transport.get_server_info()
{'node': 'riak@192.168.0.3', 'server_version': '2.1.1'}
`

**Comment by lukebakken** [comment]
@bobhall - thank you very much for that information.

**Comment by lukebakken** [comment]
@bobhall @point9repeating we've confirmed this only affects the OSS release of Riak TS:

https://github.com/basho/riak/issues/835

**Comment by seanjensengrey** [comment]
This fix is on master but didn't make it into 2.5.3

https://github.com/basho/riak-python-client/blob/2.5.3/riak/transports/tcp/transport.py#L79

I can confirm that `master` works against Riak TS OSS 1.3.0 via

```
 pip install git+https://github.com/basho/riak-python-client.git@master
```

**Comment by lukebakken** [comment]
@seanjensengrey - GitHub closes an issue when a commit or PR references the issue is committed. You can see that this particular issue is scheduled for the `2.6.0` milestone and [was not included in `2.5.3`](https://github.com/basho/riak-python-client/milestone/11?closed=1).

Anyway, let me know if this is needed soon and I'll put together `2.5.4`.

**Comment by lukebakken** [comment]
Fixed by e195e6c712634c9f0cb3a84836dd77dc5555e517
